### ISSUE TITLE
Eve's loss function

### ISSUE THREAD

**Issue by akarody** [akarody/ISTE_NeuroCrypto#5]
**Things to keep in mind**
- L1 Loss Function is used to minimize the error which is the sum of the all the absolute differences between the true value and the predicted value.
- refer https://towardsdatascience.com/life-of-alice-bob-and-eve-with-neural-net-6df0ad1d6077
- **Write legible code, include comments.** 

**Eve's loss function**
- def eve_loss(input_message, eve_out_message, batch_size):

- minimize the error in difference of bits between Eve’s output and original message using L1 norm/distance.

**Comment by mansi1501s** [comment]
def eve_loss(input_message, eve_out_message, batch_size):

  # Mapping -1 to 0 and 1 to 1
  eve_diff = (eve_out_message + 1.0)/2.0 - (input_message + 1.0)/2.0
  
  # Eve's average L1 distance Loss of the given batch
  loss = (1/batch_size)*tf.reduce_sum(tf.abs(eve_diff))
  
  return loss
